report/node.go

```diff
@@ -198,3 +198,44 @@ func (n Node) Merge(other Node) Node {
 		Children:       n.Children.Merge(other.Children),
 	}
 }
+
+// UnsafeUnMerge removes data from n that would be added by merging other,
+// modifying the original.
+// returns true if n.Merge(other) is the same as n
+func (n *Node) UnsafeUnMerge(other Node) bool {
+	// If it's not the same ID and topology then just bail out
+	if n.ID != other.ID || n.Topology != other.Topology {
+		return false
+	}
+	n.ID = ""
+	n.Topology = ""
+	remove := true
+	if n.LatestControls.EqualUnderMerge(other.LatestControls) {
+		n.LatestControls = nil
+	} else {
+		remove = false
+	}
+	if n.Latest.EqualUnderMerge(other.Latest) {
```

Comment:
Don't we really want to check if `n.Latest.ContainedIn(other.Latest)`? (same for `LatestControls`)

Reply:
If we had that method, maybe.  But that would be more complicated hence slower.  The 99% case is that nothing has changed in the 3 seconds since we last published, so I'm trying to check for that.

Reply:
As I comment above, noting that `DeepEqual` is good enough in practice most of the time would be a very useful comment in the code :)